### PR TITLE
(#149) Update Organizational Documentation for trials

### LINKED ISSUE #149: Update documentation to reflect streamlined Trial installation process
With the new licensing authentication pieces getting set up, over the next few weeks we'll be in a position to not require folks to manually install the C4B packages, Trial licenses will be usable for the licensed packages feed.

We need to update documentation to reflect this, although we can't merge those changes into the live version of the site until we have the updates to the licensed package feed up and running.

## Links

[Internal Issue](https://gitlab.com/chocolatey/licensing-administration/licensing-services/-/issues/22)

### PR DESCRIPTION
**Pull request by vexx32**
Remove mentions of trials needing to manually download packages from their trial emails; we will no longer need those steps to differ from fully licensed customers when the licensed package feed is updated.

Included a few small formatting / minor spelling fixes that I happened to catch as I was looking through.

Fixes #149

### CODE REVIEW

**Comment by gep13** [comment]
@pauby @ferventcoder are either of you in a position to review this PR?  Thanks!

**Comment by gep13** [comment]
@ferventcoder @pauby asking again for some input on this PR prior to merging in.  Thanks

**Comment by ryanrichter94** [comment]
Just a thought but can we get rid of the MSP step labels as we longer offer discrete MSP licensing?

**Comment by vexx32** [comment]
@ryanrichter94 tbh this entire guide needs rewritten / split out into multiple separate guides, at which point we'd remove the MSP sections, I think.

Probably best done in a future PR rather than this one though. 🙂

**Comment by ryanrichter94** [comment]
@vexx32 I'll raise a separate PR for work to split the Org Deployment Guide and get it in a better place. In regards to what this PR covers LGTM.

**Review by adilio — APPROVED** [review]
Hi @vexx32,

TL;DR: LGTM 🚀 

@ryanrichter94 and myself are doing some active triaging on outstanding PR's and Issues for docs that we should move forward with.

I've reviewed and approved this, and feel we should go ahead and merge this soon. As well, it is currently a logical first step on other required changes in docs related to this (e.g. Quick-Start Guide). SO we kinda need this to happen to frame our later changes.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@vexx32 @adilio @ryanrichter94 if you are all happy with the changes here, there is no requirement (unless you are looking for clarification from me) to wait on me clicking the merge button.  You should all have access to do this, so feel free to go ahead.  Once merged, and the build completes, the changes should pretty much immediately be available on the site.  If anyone has any questions about the process feel free to ask, but it has been designed that anyone on the team should be able to push out some docs changes.

For now, I have enabled auto-merging of this PR, so once the PR build completes successfully, it should automatically merge and push it out.
